vulcanai2/models/basenetwork.py

```diff
@@ -307,18 +326,49 @@ def fit(self, train_loader, val_loader, epochs, retain_graph=None, valid_interv=
 
         self._init_trainer()
 
+        self.record = dict(
+            epoch=[],
+            train_error=[],
+            train_accuracy=[],
+            validation_error=[],
+            validation_accuracy=[]
+        )
+
         try:
+            if plot is True:
+                fig_number = plt.gcf().number + 1 if plt.fignum_exists(1) else 1
+
             for epoch in trange(self.epoch, epochs, desc='Epoch: ', ncols=80):
+                epoch_time = time.time()
+
                 train_loss, train_acc = self._train_epoch()
+                train_epoch_time = time.time() - epoch_time
+
                 valid_loss, valid_acc = self._validate()
+                valid_epoch_time = time.time() - epoch_time
+
                 tqdm.write("\n Epoch {}:\n"
-                           "Train Loss: {:.6f} | Test Loss: {:.6f} | "
-                           "Train Acc: {:.4f} | Test Acc: {:.4f}".format(
+                           "Train Loss: {:.6f} | Train Acc: {:.4f} in {:.2f}s \n"
+                           "Test Loss: {:.6f} | Test Acc: {:.4f} in {:.2f}s".format(
                                 epoch,
                                 train_loss,
-                                valid_loss,
                                 train_acc,
-                                valid_acc))
+                                train_epoch_time,
```

Comment:
Why are you also putting in the time if tqdm will tell you how much time is left in that epoch?

Reply:
I would prefer to have the epoch results on one line

Reply:
Makes sense. It was there in Vulcan so I thought I would add it. I will remove

Reply:
Fair enough. yeah Vulcan's command line info wasn't as robust as tqdm which is why I had it before but this new version works better